TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -443,10 +445,111 @@ public static void CheckPicture(Node node, CommonDataDescriptionAndDataRedefines
             }
         }
 
+#if DOMAIN_CHECKER
+        /// <summary>
+        /// Get the top program.
+        /// </summary>
+        /// <param name="curPrg"></param>
+        /// <returns></returns>
+        static ProgramSymbol GetTopProgram(ProgramSymbol curPrg)
+        {
+            ProgramSymbol top = (ProgramSymbol)curPrg.TopParent(Symbol.Kinds.Program);
```

Comment:
hum... not here we are looking to the toppest program, so we must recurse.

Reply:
Yes I understand that we are looking for the top program but the whole method GetTopProgram can be replaced by a call to TopParent (with the Program kind) because TopParent already does the recursion up to the top Symbol.

Reply:
ok, right